### ISSUE TITLE
Cursor Invisible - Fedora 32 - VMware Workstation Player 15.5.6

### ISSUE THREAD

**Issue by paralllax** [swaywm/sway#5621]
- **Sway Version:**
  - `sway version 1.4`

- **Debug Log:**
  -  Weirdly, the behaviour does not reproduce when running in debug. Whether I launch `sway -d` from a gnome or sway session, the cursor will appear. In the case of when I am logged in with sway, the cursor is only visible over the debug area and not the rest of the desktop environment. I am uncertain why I can't reproduce it when calling sway manually from the terminal, versus just logging in and selecting sway for the desktop environment. 

- **Configuration File:**
  - I used the default configuration file

- **Errors in Syslog**
- These are the errors that appear (hundreds) in the syslog when attempting to use my mouse with sway. 
```
Aug 13 04:45:15 localhost.localdomain /usr/libexec/gdm-wayland-session[3792]: 2020-08-13 04:45:15 - [backend/drm/atomi
c.c:38] Atomic test failed: Invalid argument
Aug 13 04:45:15 localhost.localdomain kernel: [drm:vmw_du_cursor_plane_atomic_check [vmwgfx]] *ERROR* surface not suit
able for cursor
```

- **Description:**
  -  It is a fresh install of Fedora 32 on VMware Workstation player. Everything is the default config, it is a brand new install. I've attempted erasing and recreating the VM several times. I am uncertain if it is relevant, but when I get_inputs, I see the mouse as `VirtualPS/2 VMware VMMouse`. The cursor displays just fine with gnome, and also oddly works when I run the provided debug command, but only then... otherwise I get the errors described above.

**Comment by emersion** [comment]
Duplicate of https://github.com/swaywm/sway/issues/3814

**Comment by paralllax** [comment]
@emersion my apologies, I could have sworn I included it in my update, but it seems I forgot! This actually isn't a duplicate, I neglected to add that I had tried those fixes in the previous very similar issues, and none of them had worked. 

The following did not work/had no effect 

```
WLR_NO_HARDWARE_CURSORS=1
```

If there is any additional information I can provide, please let me know! Thank you.

**Comment by emersion** [comment]
Hmm, `WLR_NO_HARDWARE_CURSORS=1` should really help. Can you post debug logs with this env variable set?

**Comment by paralllax** [comment]
For sure, [here is the debug log](https://gist.github.com/paralllax/a7ee0478dc6d9b6de205b50c6acd5423l) when I launched it from inside gnome. Weirdly though, whenever I would launch it, the mouse cursor wouldn't disappear. 

I have `export WLR_NO_HARDWARE_CURSORS=1` set in my `~/.bash_profile`. I've also put it in `/etc/profile`

The thing that really confuses me, I found if I login with sway, then I logout and login with gnome, then logout and log back in with sway, the cursor is there. Specifically in that order. I took a dump of the env variables from when i first login with sway, then when I login again after logging in with gnome, and there are surprisingly some differences. Not sure if it helps, but figured I would include it. 

```
$ diff -i -y -w --suppress-common-lines pre post
WINDOWID=6291466					      |	SESSION_MANAGER=local/unix:@/tmp/.ICE-unix/3961,unix/unix:/tm
							      >	WINDOWID=4194314
							      >	HISTCONTROL=ignoredups
							      >	XDG_MENU_PREFIX=gnome-
I3SOCK=/run/user/1000/sway-ipc.1000.3170.sock		      |	HISTSIZE=1000
							      >	HOSTNAME=localhost.localdomain
							      >	I3SOCK=/run/user/1000/sway-ipc.1000.5029.sock
							      >	SSH_AUTH_SOCK=/run/user/1000/keyring/ssh
							      >	XMODIFIERS=@im=ibus
SWAYSOCK=/run/user/1000/sway-ipc.1000.3170.sock		      |	XDG_CURRENT_DESKTOP=GNOME
							      >	SWAYSOCK=/run/user/1000/sway-ipc.1000.5029.sock
							      >	QT_IM_MODULE=ibus
XDG_SESSION_ID=4					      |	XDG_SESSION_ID=8
							      >	MAIL=/var/spool/mail/user
```

**Comment by paralllax** [comment]
Hmm well after much more troubleshooting, I think the problem is a bug with GDM. I've tried other display managers, and the problem only exists with GDM. I tried launching sway from other TTY's and couldn't replicate the issue. For now I've just left using gdm and am using something in bashrc to detect when I am on tty1 then launch sway manually when Iogin. 

Thank you for the help! We can just leave this closed.
